services/pull/patch.go

```diff
@@ -355,23 +355,19 @@ func checkConflicts(ctx context.Context, pr *issues_model.PullRequest, gitRepo *
 	}
 
 	// 3b. Create a plain patch from head to base
-	tmpPatchFile, err := os.CreateTemp("", "patch")
+	tmpPatchFile, cleanup, err := setting.AppDataTempDir("git-repo-content").CreateTempFileRandom("patch")
 	if err != nil {
 		log.Error("Unable to create temporary patch file! Error: %v", err)
 		return false, fmt.Errorf("unable to create temporary patch file! Error: %w", err)
 	}
-	defer func() {
-		_ = util.Remove(tmpPatchFile.Name())
-	}()
+	defer cleanup()
 
 	if err := gitRepo.GetDiffBinary(pr.MergeBase+"...tracking", tmpPatchFile); err != nil {
-		tmpPatchFile.Close()
```

Comment:
Close is still necessary before cleanup?

Reply:
I don't know what you mean. Sometimes yes, sometimes no.

![image](https://github.com/user-attachments/assets/ced98766-7dff-4932-9e55-c1a7a1a700fd)

Reply:
The file is in the temp directory, so that I think it should be closed before the `cleanup` which will delete the temp directory. So that this line you removed should be necessary.

Reply:
> before the cleanup which will delete the temp directory.

I do not understand. This `cleanup` doesn't "delete the temp directory"

Reply:
It may fail because one file in this directory opened but not closed?

Reply:
WHERE IS THE DIRECTORY???!!! WHICH DIRECTORY???!!!